src/main/scala/it/pagopa/interop/apigateway/error/Handlers.scala

```diff
@@ -0,0 +1,179 @@
+package it.pagopa.interop.apigateway.error
+
+import akka.http.scaladsl.server.StandardRoute
+import com.typesafe.scalalogging.LoggerTakingImplicit
+import it.pagopa.interop.apigateway.error.GatewayErrors._
+import it.pagopa.interop.commons.logging.ContextFieldsToLog
+import it.pagopa.interop.commons.utils.errors.GenericComponentErrors.OperationForbidden
+import it.pagopa.interop.commons.utils.errors.{AkkaResponses, ServiceCode}
+
+import scala.util.{Failure, Try}
+
+object Handlers extends AkkaResponses {
+
+  implicit val serviceCode: ServiceCode = ServiceCode("013")
+
+  def handleGetAgreementError(logMessage: String, agreementId: String)(implicit
+    contexts: Seq[(String, String)],
+    logger: LoggerTakingImplicit[ContextFieldsToLog]
+  ): PartialFunction[Try[_], StandardRoute] = {
+    case Failure(ex: AgreementNotFound)          => notFound(ex, logMessage)
+    case Failure(ex: InvalidAgreementState.type) =>
+      logger.warn(s"Root cause for $logMessage", ex)
```

Comment:
Note: in similar cases we don't want to return the raised error with its description.
For example in this specific endpoint a 404 should be returned if an agreement with a not allowed state is requested.
So an additional log is written to allow us to see the underling issue.